### PR TITLE
feat(orders): add order placement and management via Client Portal API

### LINKED ISSUE #97: feat: order placement and management via Client Portal API
## Summary

Client Portal API経由でIBプラットフォームに注文を発注・変更・取消するMCPツールを実装する（Phase 2）。

## Background

Phase 1（読み取り専用）で注文確認とDB同期が完了した後、注文の発注・変更・取消機能を追加する。Claude Code/Desktopから直接IBに注文を出せるようになる。

**セキュリティ最重要**: 実資金の取引に関わるため、多重の安全機構を実装する。

### MCP Tools (`ib_sec_mcp/mcp/tools/live_trading.py` に追加)

- [ ] `place_order(account_id, symbol, side, quantity, order_type, limit_price?, ...)` — 注文発注
- [ ] `modify_order(order_id, quantity?, limit_price?)` — 注文変更
- [ ] `cancel_order(order_id)` — 注文取消
- [ ] `cancel_all_orders()` — 全注文取消（緊急用）

### Safety Mechanisms（多重安全機構）

- [ ] **確認ステップ**: IBのorder reply確認（`POST /v1/api/iserver/account/{accountId}/orders` → reply endpoint）
- [ ] **金額上限**: 1注文あたりの最大金額設定 (`IB_MAX_ORDER_AMOUNT_USD`, default: $50,000)
- [ ] **日次上限**: 1日あたりの発注総額上限 (`IB_DAILY_ORDER_LIMIT_USD`, default: $200,000)
- [ ] **発注ログ**: 全発注をローカルファイルに記録 (`data/processed/order_log.jsonl`)
- [ ] **Dry-run モード**: `IB_ORDER_DRY_RUN=1` で実際の発注を行わない（デフォルトON）
- [ ] **Read-only モード**: `IB_READ_ONLY=1` で発注系ツールを無効化
- [ ] **MCP permissions**: `settings.json` で発注系ツールを `ask` に設定する推奨設定例をドキュメントに記載

### Order Logging (`data/processed/order_log.jsonl`)

各発注レコード:
```json
{
  "timestamp": "2026-03-20T12:00:00Z",
  "action": "place|modify|cancel",
  "account_id": "U***5070",
  "symbol": "GLDM",
  "side": "BUY",
  "quantity": 25,
  "limit_price": "89.00",
  "order_id": "12345",
  "status": "submitted|rejected|dry_run",
  "dry_run": true
}
```

### Configuration

- [ ] 環境変数一覧をドキュメント化
- [ ] デフォルト: `IB_ORDER_DRY_RUN=1`（明示的にOFFにしない限り発注しない）
- [ ] `.env.example` 更新

## Acceptance Criteria

- [ ] Dry-runモードで発注APIを呼ばずにログのみ記録
- [ ] 金額上限超過時にエラー（発注しない）
- [ ] 日次上限超過時にエラー（発注しない）
- [ ] 全発注がorder_log.jsonlに記録される
- [ ] Read-onlyモードで発注系ツールが無効化される
- [ ] アカウント番号がログでマスキングされる
- [ ] ローカルDBとIBが同期される（発注後に自動sync）

## Security Considerations

- 発注系ツールは `settings.json` の `permissions.ask` に設定推奨
- Dry-runがデフォルトON — 本番発注は明示的なopt-in
- order_log.jsonl は gitignored（data/processed/ 配下）
- 環境変数のみで制御（コードにフラグをハードコードしない）

## Testing

- [ ] Unit: Dry-runモードで全ツールの動作検証
- [ ] 金額上限テスト（上限超過 → エラー）
- [ ] 日次上限テスト
- [ ] Read-onlyモード無効化テスト
- [ ] ログ記録テスト（マスキング含む）
- [ ] Integration test (manual): Paper Trading口座で発注 → 確認 → 取消

## Dependencies

- #96 CP Client (Wave 1)
- #93 MCP Live Trading Tools (Wave 2)
- #94 Order Sync (Wave 2 — 発注後の自動同期に使用)

## Post-Merge Tasks

- Required: `IB_ORDER_DRY_RUN=1` がデフォルトであることを確認
- Required: `settings.json` で発注系ツールを `ask` permission に設定
- Recommended: Paper Trading口座でDry-run → 実発注テスト
- Manual Verification: 発注 → `get_live_orders` → `get_pending_orders` で一貫性確認

### PR DESCRIPTION
**Pull request by knishioka**
## Summary

Implements order placement, modification, and cancellation via IB Client Portal Gateway API with multiple safety mechanisms (#97).

- **4 MCP tools**: `place_order`, `modify_order`, `cancel_order`, `cancel_all_orders`
- **6 safety mechanisms**: dry-run (default ON), per-order limit ($50K), daily limit ($200K), read-only mode, order logging, account masking
- **CP Client extensions**: 3 new methods + 2-step order confirmation flow
- **56 unit tests** covering all tools, safety helpers, and edge cases

## Changes

| File | Change |
|------|--------|
| `ib_sec_mcp/mcp/tools/order_management.py` | New module with 4 MCP tools + safety helpers |
| `ib_sec_mcp/api/cp_client.py` | Add `place_order`, `modify_order`, `cancel_order` methods |
| `ib_sec_mcp/api/cp_models.py` | Add `CPOrderRequest`, `CPOrderReply`, `CPOrderType` models |
| `ib_sec_mcp/mcp/tools/__init__.py` | Register order management tools |
| `tests/mcp/test_order_management.py` | 56 comprehensive tests |
| `tests/mcp/test_server.py` | Add 4 new tools to EXPECTED_TOOLS |
| `.env.example` | Document safety configuration env vars |

## Safety Design

| Mechanism | Default | Env Var |
|-----------|---------|---------|
| Dry-run mode | **ON** (no live orders) | `IB_ORDER_DRY_RUN=0` to disable |
| Per-order limit | $50,000 | `IB_MAX_ORDER_AMOUNT_USD` |
| Daily limit | $200,000 | `IB_DAILY_ORDER_LIMIT_USD` |
| Read-only mode | OFF | `IB_READ_ONLY=1` to enable |
| Order log | `data/processed/order_log.jsonl` | `IB_ORDER_LOG_PATH` |
| Account masking | Always on | N/A |

## Test plan

- [x] Dry-run mode prevents API calls (default behavior)
- [x] Amount limit exceeded returns error without placing order
- [x] Daily limit exceeded returns error without placing order
- [x] Read-only mode blocks all order tools
- [x] Account IDs masked in all log entries
- [x] All orders logged to JSONL file
- [x] Gateway connection errors handled gracefully
- [x] Session expiration handled gracefully
- [x] Invalid inputs (side, order type, quantity) rejected
- [x] Decimal precision preserved in all responses
- [x] Live order placement with 2-step confirmation
- [x] cancel_all cancels only active orders
- [ ] [manual] Paper Trading account: dry-run → live order test
- [ ] [manual] Verify `get_live_orders` → `get_pending_orders` consistency after placement

Closes #97

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ib_sec_mcp/api/cp_client.py [hunk 1]
```diff
@@ -262,6 +265,153 @@ async def get_positions(self, account_id: str) -> list[CPPosition]:
...
+            "Modifying order %d for account %s",
+            order_id,
+            mask_sensitive(account_id, show_chars=3),
+        )
+
+        payload: dict[str, object] = {}
+        if quantity is not None:
+            payload["quantity"] = float(quantity)
+        if limit_price is not None:
+            payload["price"] = float(limit_price)
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/api/cp_client.py]: Fixed in 329a835. Replaced float() with str() for Decimal serialization in modify_order payload.

File: ib_sec_mcp/api/cp_models.py [hunk 2]
```diff
@@ -98,3 +98,53 @@ class CPPosition(BaseModel):
...
+            "acctId": self.account_id,
+            "conid": self.contract_id,
+            "side": self.side.value,
+            "quantity": float(self.quantity),
+            "orderType": self.order_type.value,
+            "tif": self.tif,
+        }
+        if self.price is not None:
+            d["price"] = float(self.price)
+        return d
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/api/cp_models.py:137]: Fixed in 329a835. Replaced float() with str() in to_api_dict() for both quantity and price.

File: ib_sec_mcp/mcp/tools/order_management.py [hunk 3]
```diff
@@ -0,0 +1,781 @@
...
+                quantity=qty,
+                limit_price=price,
+                order_id=None,
+                status="rejected",
+                dry_run=False,
+            )
+            return _error_response(f"Order placement failed: {e}")
+
+    @mcp.tool
+    async def modify_order(
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/order_management.py:442]: Fixed in 329a835. Added daily limit check (check_daily_limit) to modify_order, closing the safety loophole.

File: ib_sec_mcp/api/cp_client.py [hunk 4]
```diff
@@ -262,6 +265,153 @@ async def get_positions(self, account_id: str) -> list[CPPosition]:
...
+        replies = data if isinstance(data, list) else [data]
+        result: list[CPOrderReply] = []
+        for reply in replies:
+            parsed = CPOrderReply.model_validate(reply)
+            if parsed.reply_id and not parsed.order_id:
+                confirmed = await self._confirm_order_reply(parsed.reply_id)
+                result.extend(confirmed)
+            else:
+                result.append(parsed)
+        return result
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/api/cp_client.py]: Fixed in 329a835. Extracted _parse_order_replies() helper method, used by both place_order and modify_order.

File: ib_sec_mcp/mcp/tools/order_management.py [hunk 5]
```diff
@@ -0,0 +1,781 @@
...
+
+def get_daily_total(log_path: Path) -> Decimal:
+    """Calculate total order amount placed today from order log."""
+    if not log_path.exists():
+        return Decimal("0")
+
+    today = datetime.now(UTC).strftime("%Y-%m-%d")
+    total = Decimal("0")
+
+    with open(log_path) as f:
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/order_management.py]: Fixed in 329a835. Added encoding='utf-8' to both open() calls in get_daily_total and write_order_log.

File: ib_sec_mcp/mcp/tools/order_management.py [hunk 6]
```diff
@@ -0,0 +1,781 @@
...
+                status="rejected",
+                dry_run=False,
+            )
+            return _error_response(f"Order placement failed: {e}")
+
+    @mcp.tool
+    async def modify_order(
+        account_id: str,
+        order_id: int,
+        symbol: str = "",
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/order_management.py]: Fixed in 329a835. Made symbol a required parameter in both modify_order and cancel_order for complete audit trails.

File: ib_sec_mcp/mcp/tools/order_management.py [hunk 7]
```diff
@@ -0,0 +1,786 @@
...
+            try:
+                record = json.loads(line)
+            except json.JSONDecodeError:
+                continue
+            # Only count actual submissions (not dry runs, not cancels)
+            if record.get("dry_run", False):
+                continue
+            if record.get("action") != "place":
+                continue
+            if record.get("status") != "submitted":
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/order_management.py]: Fixed in 5ca19d0. Changed status comparison to case-insensitive (.lower()) in get_daily_total() so IB replies with 'Submitted' are correctly counted toward the daily limit.

File: ib_sec_mcp/mcp/tools/order_management.py [hunk 8]
```diff
@@ -0,0 +1,786 @@
...
+        if order_type_enum is None:
+            return _error_response(
+                f"Invalid order type: {order_type}. Use 'LMT', 'MKT', 'STP', or 'STP_LMT'"
+            )
+
+        qty = Decimal(quantity)
+        if qty <= Decimal("0"):
+            return _error_response("Quantity must be positive")
+
+        price = Decimal(limit_price) if limit_price else None
```
[the remark this answers is not in the document]
Comment (knishioka) [reply ib_sec_mcp/mcp/tools/order_management.py]: Fixed in 5ca19d0. Added try/except around Decimal() conversions in place_order and modify_order. Invalid numeric inputs now return structured JSON error responses instead of crashing.
